operators/argo-controller/test/unit/test_charm.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from ops.testing import Harness
+import yaml
+
+from charm import ArgoControllerCharm
+
+
+if yaml.__with_libyaml__:
+    _DefaultDumper = yaml.CSafeDumper
+else:
+    _DefaultDumper = yaml.SafeDumper
+
+
+@pytest.fixture
+def harness():
+    return Harness(ArgoControllerCharm)
+
+
+def test_not_leader(harness):
+    harness.begin()
+    assert isinstance(harness.charm.model.unit.status, WaitingStatus)
+
+
+def test_missing_image(harness):
+    harness.set_leader(True)
+    harness.begin_with_initial_hooks()
+    assert isinstance(harness.charm.model.unit.status, BlockedStatus)
+
+
+def test_main_no_relation(harness):
+    harness.set_leader(True)
+    harness.add_oci_resource(
+        "oci-image",
+        {
+            "registrypath": "argoproj/workflow-controller:v2.3.0",
+            "username": "",
+            "password": "",
+        },
+    )
+    harness.begin_with_initial_hooks()
+    pod_spec = harness.get_pod_spec()
+
+    # confirm that we can serialize the pod spec
+    yaml.dump(pod_spec, Dumper=_DefaultDumper)
```

Comment:
Is there a reason to not just use `yaml.safe_dump`?

Reply:
You are right that would probably be better, I'll look into that and probably also change it in the interface